Animations/Animations.js

```diff
@@ -0,0 +1,194 @@
+    // componentWillEnter FUNCTION: required to for animation entering
+    // componentWillLeave FUNCTION: required to for animation leaving
+  //Additional keys
+    // componentDidEnter FUNCTION: use to define actions to run after an animation has entry completed
+    // componentDidEnter FUNCTION: use to define actions to run after an animation has leave completed
+var makeTransition = function(displayName, transitions) {
+
+  return React.createClass({
+    componentWillEnter: function(done) {
+      // This is called at the same time as componentDidMount() for components added to an existing TransitionGroup. It will block other animations from occurring until callback is called. It will not be called on the initial render of a TransitionGroup.
+      this.el = this.getDOMNode();
+      this.$el = $(this.el);
+
+      // Calls the specified function to update an animation before the next browser repaint.
+      requestAnimationFrame(function() {
+        this.$el.css(transitions.enter);
+
+        requestAnimationFrame(function() {
+          this.$el.css(transitions.enterActive);
+          Arrival(this.$el, done);
+        }.bind(this));
+      }.bind(this));
+    },
+
+    componentDidEnter: function() {
+      // This is called after the callback function that was passed to componentWillEnter is called.
+    },
+
+    componentWillLeave: function(done) {
+      // This is called after the callback function that was passed to componentWillEnter is called.
+      this.el = this.getDOMNode();
+      this.$el = $(this.el);
+
+      requestAnimationFrame(function() {
+        this.$el.css(transitions.leave);
+
+        requestAnimationFrame(function() {
+          this.$el.css(transitions.leaveActive);
+          Arrival(this.$el, done);
+        }.bind(this));
+      }.bind(this));
+    },
+
+    componentDidLeave: function() {
+      //This is called when the willLeave callback is called (at the same time as componentWillUnmount).
+    },
+
+    render: function() {
+      var { duration, ...methods } = this.props;
+      console.log(this.props);
+
+      //Change all custom props or add them if prop not defined in default
+      for(action in transitions){
+          for (css in transitions[action]){
+
+            //Duration prop work---------------------------------------
+            var duration  = this.props.duration;
+            //Use default if no duration prop defined
+            if (duration){
+              //Case to accept if duration prop is a number
+              if (typeof duration === "number") {
+                duration = duration + "ms"
+              }
+              transitions[action]['transition-duration'] = duration;
+            }
+
+            //Easing prop work------------------------------------------
+            var easing = this.props.easing;
+            //Use default if no easing prop defined
+            if (easing){
+              //Default easing cases
+              if (easing === 'easeInQuad'){
```

Comment:
Recommend changing this into an object and potentially putting it into a separate file.